### PR TITLE
deco: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
Requires
- #18209

### CODE REVIEW

File: recipes/deco/all/conanfile.py [hunk 1]
```diff
@@ -1,56 +1,81 @@
...
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 17)
+    def _min_cppstd(self):
+        return 17
 
-        minimal_version = {
+    @property
+    def _compilers_minimum_version(self):
+        return {
             "Visual Studio": "15",
```
Comment (RubenRBS) [inline recipes/deco/all/conanfile.py:32]:
```suggestion
            "Visual Studio": "15",
            "msvc": "191",
```

File: recipes/deco/all/conanfile.py [hunk 2]
```diff
@@ -1,56 +1,81 @@
...
+            "apple-clang": "9.1",
         }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("enhex-generic_serialization/1.0.0")
+        self.requires("enhex-strong_type/1.0.0")
+        self.requires("boost/1.79.0")
```
Comment (RubenRBS) [inline recipes/deco/all/conanfile.py:44]:
```suggestion
        self.requires("boost/1.82.0")
```

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot, suggesting & applying some changes before approving :)

**Comment by valgur** [comment]
@uilianries Looks like the updated recipe from #18228 has not been uploaded to ConanCenter. Can you maybe look into it?

**Comment by uilianries** [comment]
This PR requires #18209  first

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**